File: Results/DER/Translation/translation_output/no_test/gpt-4/codenet/python/java/p03502.java
Here's the equivalent Java code for the given Python program:

```java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        scanner.close();
        
        String nStr = Integer.toString(n);
        int num = 0;
        
        for (int i = 0; i < nStr.length(); i++) {
            num += Character.getNumericValue(nStr.charAt(i));
        }
        
        String ans = "No";
        
        if (n % num == 0) {
            ans = "Yes";
        }
        
        System.out.println(ans);
    }
}
```

This Java code does the following, equivalent to the Python code:
- It reads an integer from the user input.
- It converts the integer to a string to iterate over its digits.
- It sums up the digits of the number.
- It checks if the input number is divisible by the sum of its digits, and sets the answer accordingly.
- Finally, it outputs the answer.

File: Results/DER/Translation/translation_output/w_test/granite/codenet/java/python-sanitized/p03502.py
def main():
    n = int(input())
    x = n
    fx = 0
    while n >= 10:
        fx += n % 10
        n //= 10
    fx += n
    if x % fx == 0:
        print("Yes")
    else:
        print("No")

if __name__ == "__main__":
    main()